lib/timber/event.ex

```diff
@@ -13,31 +13,51 @@ defmodule Timber.Event do
 
   @doc """
   Presents the event the way it should be encoded. The log ingestion system for Timber
-  expectes events to be encoded such that their type is the key holding the event
+  expects events to be encoded such that their type is the key holding the event
   specific data.
   """
-  def event_for_encoding(nil) do
-    nil
+  def event_for_encoding(nil),
+    do: nil
+
+  def event_for_encoding(%Events.ControllerCallEvent{} = event),
+    do: event_to_map(:controller_call, event)
+
+  def event_for_encoding(%Events.CustomEvent{} = event),
+    do: event_to_map(:custom, event)
+
+  def event_for_encoding(%Events.ExceptionEvent{} = event),
+    do: event_to_map(:exception, event)
+
+  def event_for_encoding(%Events.HTTPRequestEvent{} = event),
+    do: event_to_map(:http_request, event)
+
+  def event_for_encoding(%Events.HTTPResponseEvent{} = event),
+    do: event_to_map(:http_response, event)
+
+  def event_for_encoding(%Events.SQLQueryEvent{} = event),
+    do: event_to_map(:sql_query, event)
+
+  def event_for_encoding(%Events.TemplateRenderEvent{} = event),
+    do: event_to_map(:template_render, event)
+
+  # Converts any struct into a custom event
+  def event_for_encoding(%{__struct__: name} = struct) do
+    data = Map.from_struct(struct)
+    {time_ms, data} = Map.pop(data, :time_ms)
+
+    %{custom: %{
+      name: name,
```

Comment:
This is going to be an issue. If you take a look at what comes in, it will actually be `Elixir.MyApp.MyCustomEvent`, etc. `name` will be an atom which we will probably want to convert to a string immediately and then strip off the starting `Elixir.`.